simple-schema.js

```diff
@@ -835,11 +835,14 @@ SimpleSchema.prototype.getDefinition = function(key, propList, functionContext)
 
   // For any options that support specifying a function,
   // evaluate the functions.
-  _.each(['min', 'max', 'minCount', 'maxCount', 'allowedValues', 'optional', 'label'], function (prop) {
+  _.each(['min', 'max', 'minCount', 'maxCount', 'allowedValues', 'label'], function (prop) {
     if (_.isFunction(defs[prop])) {
       defs[prop] = defs[prop].call(functionContext || {});
     }
   });
+  if (functionContext && _.isFunction(defs['optional'])) {
+    defs['optional'] = defs['optional'].call(functionContext || {});
+  }
```

Comment:
I don't do exactly what is done in the _.each. I call optional function only if `functionContext` is set.
Not the best code. It depends on fact that option prop is used only when getDefinition is called with functionContext. I didn't find a better solution without doing a bigger refactor. I didn't want to
call optional function without context. Otherwise you would have to write optional function in a way that it knows how to work without context.

Where is context not available: 
* `SimpleSchema.prototype.keyIsInBlackBox` function
* `SimpleSchema.prototype.label` function
* `SimpleSchema.prototype.messageForError` function

Maybe I missed something?